lib/katello/tasks/update_content_default_http_proxy.rake

```diff
@@ -0,0 +1,57 @@
+require File.expand_path("../engine", File.dirname(__FILE__))
+
+namespace :katello do
+  desc "Sets the content default http proxy to an existing http proxy based on supplied URL."
+  task :update_default_http_proxy => :environment do
+    setting = ::Setting::Content.where(name: 'content_default_http_proxy').first
+    options = {}
+    o = OptionParser.new
+    o.banner = "Usage: rake katello:update_content_default_http_proxy [options]"
+    o.on("-n", "--name HTTP_PROXY_NAME") do |name|
+      options[:name] = name
+    end
+    o.on("-u", "--url HTTP_PROXY_URL") do |url|
+      options[:url] = url
+    end
+    o.on("-h, --help", "Prints this help") do
+      puts o
+      exit
+    end
+    ordered_args = o.order!(ARGV) {}
+    o.parse!(ordered_args)
+
+    unless options.key?(:name)
+      $stderr.print("ERROR: Missing required option for --name HTTP_PROXY_NAME")
+      exit 2
+    end
+
+    unless options.key?(:url)
+      $stderr.print("ERROR: Missing required option for --url HTTP_PROXY_URL")
+      exit 2
+    end
+
+    User.current = User.anonymous_api_admin
+    http_proxy = HttpProxy.where(name: options[:name]).first
+
+    uri = URI(options[:url])
+    username = uri.user
+    password = uri.password
+
+    if http_proxy
+      setting.update_attribute(:value, http_proxy.name)
+      http_proxy.update_attributes(url: options[:url], username: username, password: password)
```

Comment:
since we're passing in user/pass as part of the url, i think we'd want to strip it from the url when saving it both here and below, otherwise the password is displayed as part of the url even if its not used that way in the application.

Reply:
also, anywhere we print the url, don't print the password, we don't want it appearing in logs whenever possible.

Reply:
looks like the url is still getting saved with the username and password, which we don't want.